account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -406,21 +402,25 @@ const generateSignedPermit = async <
     throw new Error("erc20Name or version is missing");
   }
 
-  let maxAmountToken = maxUint256;
+  let decimals_future = client.call({
+    to: policyToken.address,
+    data: encodeFunctionData({
+      abi: parseAbi(EIP712NoncesAbi),
+      functionName: "decimals",
+      args: [],
+    }),
+  });
 
-  if (policyToken.maxTokenAmount) {
-    let { data } = await client.call({
-      to: policyToken.address,
-      data: encodeFunctionData({
-        abi: parseAbi(EIP712NoncesAbi),
-        functionName: "decimals",
-        args: [],
-      }),
-    });
-    const decimals = 10n ** (data ? BigInt(data) : 18n);
-    maxAmountToken = policyToken.maxTokenAmount * decimals;
-  }
-  const paymasterData = await (client as Erc7677Client).request({
+  let nonce_future = client.call({
+    to: policyToken.address,
+    data: encodeFunctionData({
+      abi: parseAbi(EIP712NoncesAbi),
+      functionName: "nonces",
+      args: [account.address],
+    }),
+  });
+
+  let paymasterData_future = (client as Erc7677Client).request({
```

Comment:
❔ (no-changes) I see that it used to be casted before, do we know why it needs the casting?

Reply:
well it is because pm_getPaymasterData is only declared at Erc7677Client. https://github.com/alchemyplatform/aa-sdk/blob/82ce8ca483693137937eec1abc99583f23537839/aa-sdk/core/src/middleware/erc7677middleware.ts#L25